consensus/src/dag/bootstrap.rs

```diff
@@ -278,117 +468,47 @@ impl DagBootstrapper {
                 Ok(ack_tx) = &mut shutdown_rx => {
-                    df_handle.abort();
-                    let _ = df_handle.await;
-                    if let Err(e) = ack_tx.send(()) {
-                        error!(error = ?e, "unable to ack to shutdown signal");
-                    }
+                    shutdown_handle.shutdown().await;
+                    let _ = ack_tx.send(());
                     return;
                 },
-                sync_status = handler.run(&mut dag_rpc_rx) => {
-                    df_handle.abort();
-                    let _ = df_handle.await;
-
-                    match sync_status {
-                        StateSyncStatus::NeedsSync(certified_node_msg) => {
-                            let highest_committed_anchor_round = ledger_info_provider.get_highest_committed_anchor_round();
-                            debug!(LogSchema::new(LogEvent::StateSync).round(dag_store.read().highest_round()),
-                                target_round = certified_node_msg.round(),
-                                local_ordered_round = dag_store.read().highest_ordered_anchor_round(),
-                                local_committed_round = highest_committed_anchor_round
-                            );
-                            let dag_fetcher = DagFetcher::new(
-                                self.epoch_state.clone(),
-                                self.dag_network_sender.clone(),
-                                self.time_service.clone(),
-                                self.config.fetcher_config.clone()
-                            );
-
-                            let sync_future = sync_manager.sync_dag_to(&certified_node_msg, dag_fetcher, dag_store.clone(), highest_committed_anchor_round);
-
-                            select! {
-                                result = sync_future => {
-                                    match result {
-                                        Ok(_) => debug!("Sync finishes"),
-                                        Err(e) => error!(error = ?e, "unable to sync"),
-                                    }
-                                },
-                                Ok(ack_tx) = &mut shutdown_rx => {
-                                    let _ = ack_tx.send(());
-                                    return;
-                                }
-                            }
-
-                            debug!("going to rebootstrap.");
-                        },
-                        StateSyncStatus::EpochEnds => {
-                            // Wait for epoch manager to signal shutdown
-                            if let Ok(ack_tx) = shutdown_rx.await {
-                                let _ = ack_tx.send(());
-                            }
-                            return;
-                        },
-                        _ => unreachable!()
-                    }
+                next_mode = mode.run(&mut dag_rpc_rx, &self, &shutdown_handle) => {
```

Comment:
commented on the other PR, I don't think we need the Exit mode, just return Option and only match Some here i.e. None would close this branch